org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTDelegateCommandHandler.java

```diff
@@ -155,6 +160,14 @@ public Object executeCommand(String commandId, List<Object> arguments, IProgress
 					String proposalId = (String) arguments.get(1);
 					completionHandler.onDidCompletionItemSelect(requestId, proposalId);
 					return new Object();
+				case "java.decompile":
+					String uri = (String) arguments.get(0);
+					try {
+						byte[] bytes = Files.readAllBytes(Paths.get(new URI(uri)));
+						return DisassemblerContentProvider.getContent(bytes, monitor);
+					} catch (IOException | URISyntaxException e) {

```

Comment:
Found a problem. Create some empty file, `test.class` and open it the first time. You'll see

```
java.lang.IllegalArgumentException: 
	at org.eclipse.core.runtime.Assert.isLegal(Assert.java:66)
	at org.eclipse.core.runtime.Assert.isLegal(Assert.java:50)
	at org.eclipse.core.runtime.Status.setPlugin(Status.java:431)
	at org.eclipse.core.runtime.Status.<init>(Status.java:286)
	at org.eclipse.jdt.ls.core.internal.DisassemblerContentProvider.getContent(DisassemblerContentProvider.java:61)
	at org.eclipse.jdt.ls.core.internal.JDTDelegateCommandHandler.executeCommand(JDTDelegateCommandHandler.java:167)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:569)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
```

Have a look at how `getContent` deals with exceptions.